python-package/xgboost/core.py

```diff
@@ -91,7 +98,7 @@ def _convert_ntree_limit(
     return iteration_range
 
 
-def _expect(expectations, got):
+def _expect(expectations: Sequence[Any], got: Sequence[Any]) -> str:
```

Comment:
Maybe `Sequence[Type]`?

Reply:
Agreed; I checked the usages of `_expect` and realized `Sequence[Type]` was a better choice.